ruby/lib/transformer/syskit/extensions/system_network_generator.rb

```diff
@@ -90,6 +108,10 @@ def self.validate_frame_selection_consistency_through_inputs(task)
                                                          task_transform.to, other_transform.to)
                     end
                 end
+            rescue FrameSelectionConflict => e
+                error_handler.register_resolution_failures_from_exception(
+                    [task], e, e.message
+                )
```

Comment:
Same comment here

Bonus cleanup point if you move the sanity check itself in separate methods (e.g. find_mismatched_annotated_input and find_mismatched_transform_input)